src/main/java/Duke.java

```diff
@@ -1,10 +1,98 @@
+import java.util.Scanner;
+
+/**
+ * @author bakwxh
+ * @version 0.1
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+	/**
+	 * Storage object.
+	 */
+	private Storage storage;
+	/**
+	 * Task list object.
+	 */
+	private TaskList tasks;
+	/**
+	 * UI object.
+	 */
+	private Ui ui;
+	/**
+	 * Parser object.
+	 */
+	private Parser parser;
+
+    /**
+     * Constructor.
+     * @param filePath Path to save file.
+     */
+    public Duke(final String filePath) {
```

Comment:
I am not sure if the final keyword is needed over. The final keyword is only localised to the scope of the Duke constructor. If you are sure that there won't be any way down the function where you would accidentally modify the filePath, there is no need to finalise it.